applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -41,9 +41,148 @@ array_1d<double, NumNodes> GetPotentialOnNormalElement(const Element& rElement)
+        if (rDistances[i] > 0){
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            upper_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return upper_potentials;
+}
+
+template <int Dim, int NumNodes>
+BoundedVector<double, NumNodes> GetPotentialOnLowerWakeElement(
+    const Element& rElement, const array_1d<double, NumNodes>& rDistances)
+{
+    array_1d<double, NumNodes> lower_potentials;
+    const auto r_geometry = rElement.GetGeometry();
+
+    for (unsigned int i = 0; i < NumNodes; i++){
+        if (rDistances[i] < 0){
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+        }
+        else{
+            lower_potentials[i] = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        }
+    }
+
+    return lower_potentials;
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocity(const Element& rElement)
+{
+    const int wake = rElement.GetValue(WAKE);
+
+    if (wake == 0)
+        return PotentialFlowUtilities::ComputeVelocityNormalElement<Dim,NumNodes>(rElement);
+    else
+        return PotentialFlowUtilities::ComputeVelocityUpperWakeElement<Dim,NumNodes>(rElement);
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocityNormalElement(const Element& rElement)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(rElement.GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    data.potentials = GetPotentialOnNormalElement<Dim,NumNodes>(rElement);
+
+    return prod(trans(data.DN_DX), data.potentials);
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocityUpperWakeElement(const Element& rElement)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(rElement.GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    array_1d<double, NumNodes> distances = rElement.GetValue(ELEMENTAL_DISTANCES);
+
+    data.potentials = GetPotentialOnUpperWakeElement<Dim,NumNodes>(rElement, distances);
+
+    return prod(trans(data.DN_DX), data.potentials);
+}
+
+template <int Dim, int NumNodes>
+array_1d<double, Dim> ComputeVelocityLowerWakeElement(const Element& rElement)
+{
+    ElementalData<NumNodes, Dim> data;
+
+    // Calculate shape functions
+    GeometryUtils::CalculateGeometryData(rElement.GetGeometry(), data.DN_DX, data.N, data.vol);
+
+    array_1d<double, NumNodes> distances = rElement.GetValue(ELEMENTAL_DISTANCES);
+
+    data.potentials = GetPotentialOnLowerWakeElement<Dim,NumNodes>(rElement, distances);
+
+    return prod(trans(data.DN_DX), data.potentials);
+}
+
+template <int Dim, int NumNodes>
+double ComputePressureCoefficient(const Element& rElement, const ProcessInfo& rCurrentProcessInfo)
```

Comment:
Should this be kept in the element? Since this function will be different for the compressible element.

Or were you planning to add a ComputeCompressiblePressureCoefficient in the utilities?

Reply:
I will follow your suggestion and change the name of this function to ComputeIncompressiblePressureCoefficient. 

I think we should not keep this function in the element for two reasons: it does not access private members and having it in the utilities allows to easily test it.